### ISSUE TITLE
Feature request - Add `type` prop to `Button` component

### ISSUE THREAD

**Issue by cyruskorn** [tremorlabs/tremor-npm#205]
When using the `Button` component inside am HTML form, I can't set the button `type` to `'submit'`. Giving access to the `type` prop of the button element through to `Button` component would make this possible with very little change.

Example use:
```jsx
<form onSubmit={handleSubmit}>
  ...  
  <Button type='submit' text='Create item' />
</form>
```

**Comment by severinlandolt** [comment]
Heya @cyruskorn, thank you for using our library. This feature makes a lot of sense, great that you reached out✨

**Comment by severinlandolt** [comment]
Related to #188

**Comment by mitrotasios** [comment]
fixed in `v1.3.0` #240
